flake.nix

```diff
@@ -0,0 +1,89 @@
+{
+  description = "Nix flake for zkLLVM";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+    nix-3rdparty = {
+      url = "github:NilFoundation/nix-3rdparty";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+    nil-crypto3 = {
+      url = "https://github.com/NilFoundation/crypto3?submodules=0&rev=caf78b83692b39f504c4ee74a3834a4c38662a77";
+      type = "git";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+      };
+    };
+    nil-zkllvm-blueprint = {
+      url = "https://github.com/NilFoundation/zkllvm-blueprint";
+      type = "git";
+      submodules = true;
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        nil-crypto3.follows = "nil-crypto3";
+      };
+    };
+  };
+
+  outputs = { self
+    , nixpkgs
+    , flake-utils
+    , nix-3rdparty
+    , nil-crypto3
+    , nil-zkllvm-blueprint
+    }:
+    (flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [ nix-3rdparty.overlays.${system}.default ];
+        };
+        crypto3 = nil-crypto3.packages.${system}.default;
+        blueprint = nil-zkllvm-blueprint.packages.${system}.default;
+
+      in {
+        packages = rec {
+          zkllvm = (pkgs.callPackage ./zkllvm.nix { 
+            src_repo = self;
```

Comment:
```nix
inherit pkgs, crypto3, blueprint;
```

Reply:
I meant that when passing arguments to `pkgs.callPackage`, you can use `inherit`. Instead of this:

```nix
pkgs.callPackage ./zkllvm.nix { 
    src_repo = self;
    pkgs=pkgs;
    crypto3 = crypto3;
    blueprint = blueprint;
}
```

you can write this:

```nix
pkgs.callPackage ./zkllvm.nix { 
    src_repo = self;
    inherit crypto3, blueprint;
}
```

As far as I know, you don't need to pass `pkgs` explicitly, it will be passed automatically by `callPackage`.

Reply:
done